pkg/statsd/statsd.go

```diff
@@ -138,30 +145,30 @@ func (s *Server) RunWithCustomSocket(ctx context.Context, sf SocketFactory) erro
 		}
 	}()
 
-	receiver := NewMetricReceiver(s.Namespace, s.IgnoreHost, handler, statser, s.ReceiveBatchSize)
+	receiver := NewMetricReceiver(s.Namespace, s.IgnoreHost, metrics, events, statser, s.ReceiveBatchSize)
 	stage.StartWithContext(receiver.RunMetrics)
 	for r := 0; r < s.MaxReaders; r++ {
 		stage.StartWithContext(func(ctx context.Context) {
 			receiver.Receive(ctx, c)
 		})
 	}
 
-	// 6. Start the Flusher
-	flusher := NewMetricFlusher(s.FlushInterval, dispatcher, handler, s.Backends, ip, hostname, statser)
+	// 8. Start the Flusher
+	flusher := NewMetricFlusher(s.FlushInterval, backendHandler, s.Backends, ip, hostname, statser)
 	stage.StartWithContext(flusher.Run)
 
-	// 7. Send events on start and on stop
+	// 9. Send events on start and on stop
 	// TODO: Push these in to statser
-	defer sendStopEvent(handler, ip, hostname)
-	sendStartEvent(ctx, handler, ip, hostname)
+	defer sendStopEvent(events, ip, hostname)
+	sendStartEvent(ctx, events, ip, hostname)
 
-	// 8. Listen until done
+	// 10. Listen until done
```

Comment:
leave it here, but I might do away with this numbering in my PR... don't really feel it adds much except diff noise

Reply:
The stager is really nice and lets us pull almost everything out to functions (previously there was a heap of defer calls).  That would be a good cleanup to address both the function length and numbering in one.